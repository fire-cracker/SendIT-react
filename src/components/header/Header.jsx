import React, { Component } from 'react';
import { NavLink } from 'react-router-dom';
import { ToastContainer } from 'react-toastify';

import './style.css';

export class Header extends Component {
  state = {
    showLogin: true,
  };

toggleIcon = () => {
  this.setState(prevState => ({
    showLogin: !prevState.showLogin
  }));
};


render() {
  const { showLogin } = this.state;
  return (
    <div id="header" className="header">
      <ToastContainer />
      <header id="navBar" className={showLogin ? 'navBar' : 'responsive--navBar'}>
        <div id="headerContent">

          <div className={showLogin ? 'navigation' : 'navigation responsive'} id="navigation">
            <NavLink
              className="logout"
              id="logout"
              to="/"
            >
              Logout
            </NavLink>
            <NavLink
              className="prefile"
              id="profile"
              to="/profile"
            >
             Profile
            </NavLink>
            <NavLink
              className="create__order"
              id="create__order"
              to="/createOrder"
            >
            Create Order
            </NavLink>
            <NavLink
              className="track"
              id="track"
              to="/track"
            >
            Track
            </NavLink>
            <a
              className="icon"
              id="icon"
              onClick={this.toggleIcon}
            >
              <i className="fa fa-bars" />
            </a>
          </div>
        </div>

        <div className={showLogin ? 'blink_caption' : 'responsive--blink_caption'} href="home.html">
          <div id="blin" className="blink">Just SendIT!!!</div>
        </div>
      </header>
    </div>
  );
}
}

export default Header;
