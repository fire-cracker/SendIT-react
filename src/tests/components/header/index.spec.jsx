import React from 'react';
import { shallow } from 'enzyme';

import Header from '../../../components/header';

let wrapper;

beforeEach(() => {
  wrapper = shallow(<Header />);
});
describe('Tests for the Header', () => {
  it('Should render Header', () => {
    wrapper = shallow(<Header />);
    expect(wrapper.length).toBe(1);
    expect(wrapper).toMatchSnapshot();
  });

  it('Should call handle functions onclick', () => {
    const wrapper = shallow(<Header />);
    wrapper.find('a').at(0).simulate('click');
    wrapper.unmount();
  });
});
