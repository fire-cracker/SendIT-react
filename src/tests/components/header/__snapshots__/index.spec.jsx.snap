// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for the Header Should render Header 1`] = `
<div
  className="header"
  id="header"
>
  <ToastContainer
    autoClose={5000}
    bodyClassName={null}
    className={null}
    closeButton={
      <CloseButton
        ariaLabel="close"
      />
    }
    closeOnClick={true}
    draggable={true}
    draggablePercent={80}
    hideProgressBar={false}
    newestOnTop={false}
    pauseOnFocusLoss={true}
    pauseOnHover={true}
    position="top-right"
    progressClassName={null}
    progressStyle={null}
    rtl={false}
    style={null}
    toastClassName={null}
    transition={[Function]}
  />
  <header
    className="navBar"
    id="navBar"
  >
    <div
      id="headerContent"
    >
      <div
        className="navigation"
        id="navigation"
      >
        <Link
          id="logout"
          replace={false}
          to="/"
        >
          Logout
        </Link>
        <Link
          id="profile"
          replace={false}
          to="/profile"
        >
          Profile
        </Link>
        <Link
          id="active"
          replace={false}
          to="/createOrder"
        >
          Create Order
        </Link>
        <Link
          id="track"
          replace={false}
          to="/track"
        >
          Track
        </Link>
        <a
          className="icon"
          id="icon"
          onClick={[Function]}
        >
          <i
            className="fa fa-bars"
          />
        </a>
      </div>
    </div>
    <div
      className="blink_caption"
      href="home.html"
    >
      <div
        className="blink"
        id="blin"
      >
        Just SendIT!!!
      </div>
    </div>
  </header>
</div>
`;
